Fix integration tests

alfredeen opened the pull request:
## Description

This PR fixes two failing integration tests. One test lacked proper visibility verifications around a model and nested button. The other test failed caused by missing required app description. Also added many should('be.visible') clauses to make tests more robust and also cleaned up some comments. I also converted one obsolete test to test that redeploy of shiny apps works in regards to the app status when changing the docker image. This test may fail until the app status code fixes this.

For some background, see Jira [link](https://scilifelab.atlassian.net/browse/SS-1254)

## Checklist

- [x] This pull request is against **develop** branch (not applicable for hotfixes)
- [ ] I have included a link to the issue on GitHub or JIRA (if any)
- [ ] I have included migration files (if there are changes to the model classes)
- [x] I have included, reviewed and executed tests (unit and end2end) to complement my changes
- [x] I have updated the related documentation (if necessary)
- [x] I have added a reviewer for this pull request
- [x] I have added myself as an author for this pull request
- [ ] In the case I have modified settings.py, then I have also updated the studio-settings-configmap.yaml file in serve-charts

churnikov reviewed (commented):
I'll be honest, I did not really read into it, because it seems like it's mostly style changes and added `.should("be.visible")` calls.
So I don't have much to say here, and if tests pass, then lgtm. 

One request though, I don't see in the readme instructions on how to run integration tests. Could we add it there?

alfredeen commented:
> I'll be honest, I did not really read into it, because it seems like it's mostly style changes and added `.should("be.visible")` calls. So I don't have much to say here, and if tests pass, then lgtm.

Yes there are many such changes. I also added a test that reproduces some shiny proxy app status problems. This test may no intentionally fail before a future fix is in place.

> One request though, I don't see in the readme instructions on how to run integration tests. Could we add it there?

I see that our README has no mention of tests at all. I will add a section and mention this.

churnikov commented on README.md:
```diff
@@ -172,6 +172,52 @@ To check, go to  PyCharm | Settings | Languages & Frameworks | Django and check
+
+### End2End tests
+
+UI end2end tests and Integration tests use the [Cypress](https://www.cypress.io/) framework and are implemented in javascript. The UI end2end are automatically executed as part of git workflows. You may also run them manually in this manner after first installing Cypress on your machine:
```
I think it's worth mentioning that this requires a running instance of serve on a local machine.
